firmware/Core/Src/telecommands/eps_telecommands.c

```diff
@@ -520,3 +520,82 @@ uint8_t TCMDEXEC_eps_get_piu_housekeeping_data_run_avg_json(
     return 0;
 }
 
+uint8_t TCMDEXEC_eps_get_list_of_enabled_channels(
+    const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+    char *response_output_buf, uint16_t response_output_buf_len
+) {
+
+    EPS_struct_piu_housekeeping_data_eng_t status;
+    const uint8_t result = EPS_CMD_get_piu_housekeeping_data_eng(&status);
+    if (result != 0) {
+        snprintf(response_output_buf, response_output_buf_len,
+            "EPS_CMD_get_piu_housekeeping_data_run_avg (err %d)", result);
+        return 1;
+    }
+     
+    uint16_t status_bitfield = status.stat_ch_on_bitfield;
+    uint16_t status_oc_bitfield = status.stat_ch_overcurrent_fault_bitfield;
+    strcat(response_output_buf, "Current Channels Enabled: ");
+    for (int i =0; i<16; i++){ // check bit field for channels 0-15 
+        uint8_t bit_status = (status_bitfield >> i) & 1; // bit shift right and then check bit status
+        if (bit_status == 1) {
+            switch (i) {
+                case EPS_CHANNEL_VBATT_STACK:
```

Comment:
Please refactor this switch statement into a function (if it does not already exist) which converts it to just a string. 

Please put it in `Core/Src/eps_drivers/eps_channel_control.c`. See that file for an example of the string conversion. Consider using ChatGPT with clever prompting to invert the existing switch statement there.